toolchain/lower/constant.cpp

```diff
@@ -210,13 +210,18 @@ static auto EmitAsConstant(ConstantContext& context, SemIR::IntValue inst)
 
   // IntLiteral is represented as an empty struct. All other integer types are
   // represented as an LLVM integer type.
-  if (!llvm::isa<llvm::IntegerType>(type)) {
+  auto* int_type = llvm::dyn_cast<llvm::IntegerType>(type);
+  if (!int_type) {
     auto* struct_type = llvm::dyn_cast<llvm::StructType>(type);
     CARBON_CHECK(struct_type && struct_type->getNumElements() == 0);
     return llvm::ConstantStruct::get(struct_type);
   }
 
-  return llvm::ConstantInt::get(type, context.sem_ir().ints().Get(inst.int_id));
+  auto val = context.sem_ir().ints().Get(inst.int_id);
+  int bit_width = int_type->getBitWidth();
+  bool is_signed = context.sem_ir().GetIntTypeInfo(inst.type_id).is_signed;
```

Comment:
To the comment in type.h about IsSignedInt maybe being helpful, this could be:

```suggestion
  bool is_signed = context.sem_ir().types().IsSignedInt(inst.type_id);
```

This may even be positive now, since the bit width of `inst.type_id` isn't used here, you're getting it from `int_type`

Reply:
Because I was touching this code anyways, I tried to use `GetIntTypeInfo` consistently rather than adding more calls to `IsSignedInt`.

I'm happy to iterate on what code we want here. I had expected us to want to get rid of this entirely and be able to get `is_signed` from the `int_type` eventually. I can leave a TODO to that effect? (I just didn't want to get into improving helpers on `IntType` in this PR.)